tests/ut/azure_iot_provisioning_client_ut.c

```diff
@@ -595,95 +633,133 @@ static void testAzureIoTProvisioningClient_Register_RegistrationFailure( void **
+    prvRegister ( &xTestProvisioningClient );
 
-    /* Read response */
+    /* Publish Registration Query */
+    will_return( AzureIoTMQTT_Publish, eAzureIoTMQTTSuccess );
     will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    xPacketInfo.ucType = 0;
     assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient,
                                                            azureiotprovisioningNO_WAIT ),
                       eAzureIoTErrorPending );
 
-    /* Wait for timeout */
-    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
-    ullUnixTime += 2;
+    /* Invalid registration response */
+    prvGenerateShortFailureResponse( &xPublishInfo );
     assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient,
                                                            azureiotprovisioningNO_WAIT ),
                       eAzureIoTErrorPending );
 
-    /* Publish Registration Query */
-    will_return( AzureIoTMQTT_Publish, eAzureIoTMQTTSendFailed );
+    /* Process Invalid response */
     assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient,
                                                            azureiotprovisioningNO_WAIT ),
-                      eAzureIoTErrorPublishFailed );
-
+                      eAzureIoTErrorServerError );
     AzureIoTProvisioningClient_Deinit( &xTestProvisioningClient );
 }
 /*-----------------------------------------------------------*/
 
-static void testAzureIoTProvisioningClient_Register_InvalidResponseFailure( void ** ppvState )
+static void testAzureIoTProvisioningClient_Register_QueryFailure( void ** ppvState )
 {
     AzureIoTProvisioningClient_t xTestProvisioningClient;
     AzureIoTMQTTPublishInfo_t xPublishInfo;
 
     ( void ) ppvState;
 
     prvSetupTestProvisioningClient( &xTestProvisioningClient );
+    prvRegister ( &xTestProvisioningClient );
 
-    /* Connect */
-    prvRegistrationConnectStep( &xTestProvisioningClient );
+    /* Publish Registration Query */
+    will_return( AzureIoTMQTT_Publish, eAzureIoTMQTTSuccess );
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    xPacketInfo.ucType = 0;
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient,
+                                                           azureiotprovisioningNO_WAIT ),
+                      eAzureIoTErrorPending );
 
-    /* Subscribe */
-    prvRegistrationSubscribeStep( &xTestProvisioningClient );
+    /* Invalid registration response */
+    prvGenerateFailureResponse( &xPublishInfo );
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient,
+                                                           azureiotprovisioningNO_WAIT ),
+                      eAzureIoTErrorPending );
 
-    /* AckSubscribe */
-    prvRegistrationAckSubscribeStep( &xTestProvisioningClient );
+    /* Process Invalid response */
```

Comment:
nit here and elsewhere
```suggestion
    /* Process invalid response */
```